src/components/form/form.ts

```diff
@@ -1,24 +1,137 @@
 import Block from 'core/Block';
 
 import './form.scss';
-
+import {validateForm, ValidateRuleType} from "../../asserts/utils/validateForm";
+import {login} from "../../services/auth";
+import {formatBytes} from "../../asserts/utils";
+export type SendData = {
+    data: Record<string, string>
+    form: Record<string, string | Array<[Record<string, string>]>>
+}
 
 interface FormProps {
     form: {};
     onSubmit: any;
     onFocus: any;
     onBlur: any;
     onInput: any;
-    onChange: any;
     loginValue: string;
+    errorAddClass?: string;
+    file?: boolean;
     profileMainPage?: boolean;
 }
 
 export class Form extends Block {
     static componentName = 'Form';
+    private formCollection: HTMLCollection | object | undefined;
+    private formElems: Record<string, HTMLInputElement> | undefined;
+    private formRefs: { [p: string]: Block; } | undefined;
+    private formButton: HTMLButtonElement | undefined;
+    private formError: Block | undefined;
+    private form: HTMLFormElement | undefined;
+
+    constructor({...props}: FormProps) {
+        super({...props});
+        this.setProps({
+            // store: Store.instance(),
+            onSubmitForm: (event: MouseEvent) => this.onSubmitForm(event),
+            events: {
+
+                change: {
+                    fn: this.onChangeFile.bind(this),
+                    options: false
+                }
+            }
+        })
+    }
+
+    onChangeFile(event: Event) {
+        this.elemInit();
+        const input = event.target as HTMLInputElement
+        if (!input.type || input.type !== 'file' || !input.files) return;
+
+        let errorName = '';
+        this.formError && this.formError.props.errorName && this.formError.setProps({errorName})
+
+        if (this.form && this.form.file.files && this.form.file.files[0]) {
+            const file = this.form.file.files[0];
+            const size = file.size;
+            if (size > 1048576) {
```

Comment:
магические константы лучше помещать в именованные целочисленные константы 
```suggestion
            const MAX_FILE_SIZE = 1048576
            if (size > MAX_FILE_SIZE) {
```

Reply:
добавил константу